### ISSUE TITLE
Better error handling wrt syntax errors

### ISSUE THREAD

**Issue by Jazzpirate** [UniFormal/MMT#241]
Issues with syntactically broken mmt files:
- theories that include themselves (possibly also longer circular dependency chains) cause the solver to throw stack overflows, resulting in jEdit giving no feedback whatsoever.
- typos in declarations leading to unknown variables cause the solver to seemingly freeze, apparently especially in defined constants
...

**Comment by Jazzpirate** [comment]
- the latter seems to have no connection to subtyping :P debugger always seems to hang in methods related to unsolved variables, never in the prover or in checkSubtyping
- Typos in module paths (e.g. includes) never disappear in jedit; one needs to clear everytime one occurs to get rid of it

**Comment by florian-rabe** [comment]
Indeed, the cyclic dependency check has never been fully implemented.
It should be revisited as part of full review of the Library and StructureChecker.
